lib/internal/modules/cjs/loader.js

```diff
@@ -983,8 +982,17 @@ Module.prototype.load = function(filename) {
 
   const extension = findLongestRegisteredExtension(filename);
   // allow .mjs to be overridden
-  if (StringPrototypeEndsWith(filename, '.mjs') && !Module._extensions['.mjs'])
-    throw new ERR_REQUIRE_ESM(filename);
+  if (StringPrototypeEndsWith(filename, '.mjs') &&
+    !Module._extensions['.mjs']) {
+    process.emitWarning('Attempting to require and ESModule.' +
+                        'Replace `require` with `import`',
+                        'Warning', 'WARN_REQUIRE_ESM');
+    const ESMLoader = asyncESM.ESMLoader;
+    const exports = ESMLoader.import(filename);
```

Comment:
I think this needs to be converted to a URL for windows paths.

Reply:
It also makes sense to expose the inners of `getModuleJob` directly (namely all the bits after the specifier is resolved, like in the other PR) - the cjs loader has already resolved the specifier at this point so there's no reason to have the esm loader double resolve it. (That's just more unneeded disk hits, and potential format confusion bugs if the two loaders disagree on the format)